src/main.ts

```diff
@@ -30,6 +35,48 @@ export default function vitePluginArraybuffer(): PluginOption {
           ${new Uint8Array(await promises.readFile(file)).join(',')}
         ])`;
       }
+      if (id.endsWith("?arraybuffer&base64")) {
+        const file = id.slice(0, -12)
+        this.addWatchFile(file)
+
+        const buffer = await promises.readFile(file)
+        const b64 = buffer.toString("base64")
+
+        return `
+          function toUint8(b64) {
```

Comment:
Here instead of a simple base64 decoder (`atob` is planned to be removed) we can use something like `import decode64 from 'virtual:decode-64'` and we will generate Another function to resolve module 'virtual:decode-64'

```ts

function b64ToUint6(nChr: number) {
  return nChr > 64 && nChr < 91
    ? nChr - 65
    : nChr > 96 && nChr < 123
    ? nChr - 71
    : nChr > 47 && nChr < 58
    ? nChr + 4
    : nChr === 43
    ? 62
    : nChr === 47
    ? 63
    : 0
}

export function base64ToUint8(sBase64: string, nBlocksSize?: number) {
  const sB64Enc = sBase64.replace(/[^A-Za-z0-9+/]/g, "")
  const nInLen = sB64Enc.length
  const nOutLen = nBlocksSize
    ? Math.ceil(((nInLen * 3 + 1) >> 2) / nBlocksSize) * nBlocksSize
    : (nInLen * 3 + 1) >> 2
  const taBytes = new Uint8Array(nOutLen)

  let nMod3
  let nMod4
  let nUint24 = 0
  let nOutIdx = 0
  for (let nInIdx = 0; nInIdx < nInLen; nInIdx++) {
    nMod4 = nInIdx & 3
    nUint24 |= b64ToUint6(sB64Enc.charCodeAt(nInIdx)) << (6 * (3 - nMod4))
    if (nMod4 === 3 || nInLen - nInIdx === 1) {
      nMod3 = 0
      while (nMod3 < 3 && nOutIdx < nOutLen) {
        taBytes[nOutIdx] = (nUint24 >>> ((16 >>> nMod3) & 24)) & 255
        nMod3++
        nOutIdx++
      }
      nUint24 = 0
    }
  }

  return taBytes
}

```

Reply:
This may not be worth the added complexity. `atob` used to be marked as deprecated, but [it was determined that the deprecation was incorrect](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/65494). It was originally removed from node, but [readded in version 16](https://github.com/nodejs/node/issues/3462).

Vite's own base64 implementations take advantage of `atob`. Perhaps it make sense to copy/paste [their implementation](https://github.com/vitejs/vite/blob/1dccc3713a383ac274a36dfcaabcaaa88e380bd1/packages/vite/src/node/plugins/worker.ts#L293):

```
const decodeBase64 = (base64) => Uint8Array.from(atob(base64), c => c.charCodeAt(0));
```